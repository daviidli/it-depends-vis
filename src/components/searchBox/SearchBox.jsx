import React from 'react';
import PropTypes from 'prop-types';
import { Input, AutoComplete } from 'antd';
import { AiOutlineSearch } from 'react-icons/ai';
import {
	pipe, map, objOf, mergeRight, of
} from 'ramda';

import './SearchBox.scss';

const createLabelObj = value => ({ value, label: <div className="searchItem">{ value }</div> });

const createOptions = pipe(
	map(createLabelObj),
	objOf('options'),
	mergeRight({ label: 'Files' }),
	of
);

const SearchBox = ({ files, setSelectedFile }) => (
	<AutoComplete
		autoFocus
		filterOption
		options={createOptions(files)}
		dropdownClassName="searchDropdown"
		dropdownMatchSelectWidth={400}
		onSelect={setSelectedFile}
	>
		<Input
			className="searchInput"
			size="large"
			placeholder="Select a file"
			prefix={
				<AiOutlineSearch size={24} color="grey" className="searchIcon" />
			}
		/>
	</AutoComplete>
);

SearchBox.propTypes = {
	setSelectedFile: PropTypes.func.isRequired,
	files: PropTypes.arrayOf(PropTypes.string)
};

SearchBox.defaultProps = {
	files: []
};

export default SearchBox;
