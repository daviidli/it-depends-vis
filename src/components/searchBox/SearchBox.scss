.searchIcon {
	padding-top: 8px;
}

.ant-select-selection-search-input {
	border-radius: 100em;
	height: 42px;
	width: 95% !important;
	margin: 0 2px;
	background-color: #efefef;
	outline: none !important;
	border: none !important;
	box-shadow: none !important;
}

.ant-page-header-heading {
	.ant-input {
		background-color: #efefef;
	}
}
