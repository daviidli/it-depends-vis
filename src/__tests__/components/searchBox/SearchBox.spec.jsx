import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { clone } from 'ramda';
import SearchBox from '../../../components/searchBox/SearchBox';

const defaultProps = {
	setSelectedFile: jest.fn(),
	files: ['file1', 'file2']
};

const setup = (props = defaultProps) => {
	const wrapper = shallow(<SearchBox {...props} />);
	return { wrapper };
};

describe('SearchBox component', () => {
	it('should match snapshot', () => {
		const { wrapper } = setup();
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should match snapshot with no files prop', () => {
		const props = clone(defaultProps);
		props.files = undefined;
		const { wrapper } = setup();
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
