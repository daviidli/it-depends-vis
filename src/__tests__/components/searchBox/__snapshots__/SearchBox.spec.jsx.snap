// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox component should match snapshot 1`] = `
<ForwardRef(AutoComplete)
  autoFocus={true}
  dropdownClassName="searchDropdown"
  dropdownMatchSelectWidth={400}
  filterOption={true}
  onSelect={[MockFunction]}
  options={
    Array [
      Object {
        "label": "Files",
        "options": Array [
          Array [
            Object {
              "label": <div
                className="searchItem"
              >
                file1
              </div>,
              "value": "file1",
            },
            Object {
              "label": <div
                className="searchItem"
              >
                file2
              </div>,
              "value": "file2",
            },
          ],
        ],
      },
    ]
  }
>
  <Input
    className="searchInput"
    placeholder="Select a file"
    prefix={
      <AiOutlineSearch
        className="searchIcon"
        color="grey"
        size={24}
      />
    }
    size="large"
    type="text"
  />
</ForwardRef(AutoComplete)>
`;

exports[`SearchBox component should match snapshot with no files prop 1`] = `
<ForwardRef(AutoComplete)
  autoFocus={true}
  dropdownClassName="searchDropdown"
  dropdownMatchSelectWidth={400}
  filterOption={true}
  onSelect={[MockFunction]}
  options={
    Array [
      Object {
        "label": "Files",
        "options": Array [
          Array [
            Object {
              "label": <div
                className="searchItem"
              >
                file1
              </div>,
              "value": "file1",
            },
            Object {
              "label": <div
                className="searchItem"
              >
                file2
              </div>,
              "value": "file2",
            },
          ],
        ],
      },
    ]
  }
>
  <Input
    className="searchInput"
    placeholder="Select a file"
    prefix={
      <AiOutlineSearch
        className="searchIcon"
        color="grey"
        size={24}
      />
    }
    size="large"
    type="text"
  />
</ForwardRef(AutoComplete)>
`;
